### DISCUSSION TITLE
"Missing required native library dependency: libsignal-client"

### DISCUSSION THREAD

**Discussion by keysmashes** [AsamK/signal-cli#601] [Q&A] [answered]
Whenever I try to e.g. register an account, I get an error:

```console
$ signal-cli --verbose -u ... register
2021-04-19T22:51:54.772+0100 [main] INFO LibSignal - [libsignal-client]: rust/bridge/jni/src/logging.rs:173: Initializing libsignal-client version:0.4.0
Missing required native library dependency: libsignal-client
```

I've followed the instructions on the wiki about building the native libraries, since I'm using a Raspberry Pi 4.

AIUI, the lack of a "Support for new group V2 is disabled" message means that it's successfully found libzkgroup (since it [checks for that first](https://github.com/AsamK/signal-cli/blob/3935ebf81bf9ccdb51c0b62b0bfda8972020238d/src/main/java/org/asamk/signal/App.java#L122)), which is supported by the fact that it literally prints [a log message from libsignal-client](https://github.com/signalapp/libsignal-client/blob/22ff571c624dd9a47d37d784ba872669140071af/rust/bridge/jni/src/logging.rs#L173-L176) before it gives up.

So why on earth is this not working?

**Comment by keysmashes** [comment, marked as answer]
the answer: I was using a too-new version of libsignal-client – you must use a version from before [this commit](https://github.com/signalapp/libsignal-client/commit/26d46e3d2d3304c1b31a99d76a24990a79c96b0c), the best version is probably the tag `java-0.2.3`, since that matches the bundled one afaik.

**Comment by chenrui333** [comment]
Yeah, I have run into the similar issue with [the homebrew build](https://github.com/Homebrew/homebrew-core/pull/79849), the version scheme change is indeed the root cause.
